src/main/scala/org/coroutines/AstCanonicalization.scala

```diff
@@ -122,6 +121,48 @@ trait AstCanonicalization[C <: Context] {
       (decls, q"$localvarname")
     case q"$selector[..$tpts](...$paramss)" if tpts.length > 0 || paramss.length > 0 =>
       // application
+
+      // We first need to see which parameters, if any, are by-name. Those that are
+      // should not be canonicalized.
+
+      // Maps to the parameter lists, saying whether or not they are by-name.
+      val byNameParams: immutable.Seq[immutable.Seq[Boolean]] = {
+        // Check that `selector` has a non-trivial symbol. If it doesn't, we assume
```

Comment:
You're right: non-trivial isn't the best way to describe what I meant.

Some methods threw errors when I called `asMethod` on them. These checks are my way of preventing those exceptions.

If I recall correctly, the problematic functions were lambdas.